geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningDUnitTest.java

```diff
@@ -447,72 +454,62 @@ public void testPut_ValidateDataOnMember_OnlyPrimary_Accessor() {
     member3.invoke(() -> FixedPartitioningTestBase.setPRObserverBeforeCalculateStartingBucketId());
-    member4.invoke(() -> FixedPartitioningTestBase.setPRObserverBeforeCalculateStartingBucketId());
     try {
 
+      member4.invoke(() -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter",
+          null, 0, 0, 12, new QuarterPartitionResolver(), null, false));
+
       FixedPartitionAttributes fpa1 =
           FixedPartitionAttributes.createFixedPartition(Quarter1, true, 3);
-      List<FixedPartitionAttributes> fpaList = new ArrayList<FixedPartitionAttributes>();
-      fpaList.add(fpa1);
       AsyncInvocation inv1 = member1.invokeAsync(
-          () -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter", fpaList, 0,
+          () -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter",
+              Arrays.asList(fpa1), 0,
               40, 12, new QuarterPartitionResolver(), null, false));
 
       FixedPartitionAttributes fpa2 =
           FixedPartitionAttributes.createFixedPartition(Quarter2, true, 3);
-      fpaList.clear();
-      fpaList.add(fpa2);
       AsyncInvocation inv2 = member2.invokeAsync(
-          () -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter", fpaList, 0,
+          () -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter",
+              Arrays.asList(fpa2), 0,
               40, 12, new QuarterPartitionResolver(), null, false));
 
       FixedPartitionAttributes fpa3 =
           FixedPartitionAttributes.createFixedPartition(Quarter3, true, 3);
-      fpaList.clear();
-      fpaList.add(fpa3);
       AsyncInvocation inv3 = member3.invokeAsync(
-          () -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter", fpaList, 0,
+          () -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter",
+              Arrays.asList(fpa3), 0,
               40, 12, new QuarterPartitionResolver(), null, false));
 
-      member4.invoke(() -> FixedPartitioningTestBase.createRegionWithPartitionAttributes("Quarter",
-          null, 0, 0, 12, new QuarterPartitionResolver(), null, false));
-      try {
-        member4.invoke(() -> FixedPartitioningTestBase.putThorughAccessor_Immediate("Quarter"));
-      } catch (Exception e) {
-        e.printStackTrace();
-        if (!(e.getCause() instanceof PartitionNotAvailableException)) {
-          Assert.fail("exception thrown is not PartitionNotAvailableException", e);
-        }
-      }
-      try {
-        inv1.join();
-        inv2.join();
-        inv3.join();
-      } catch (InterruptedException e) {
-        e.printStackTrace();
-        Assert.fail("Unexpected Exception", e);
-      }
+      getBlackboard().waitForGate("waiting", 1, TimeUnit.MINUTES);
+
+      Assertions.assertThatThrownBy(() -> member4
+          .invoke(() -> FixedPartitioningTestBase.putThorughAccessor_Immediate("Quarter")))
```

Comment:
typo in method name here

Reply:
Also, which member is the data being put to, what state is it in (blocked at a gate, or blocked by a DLock?), and what happens if that changes?